core-java/src/main/java/eu/ostrzyciel/jelly/core/JellyOptions.java

```diff
@@ -0,0 +1,197 @@
+        int systemSupportedVersion
+    ) {
+        if (
+            requestedOptions.getVersion() > supportedOptions.getVersion() ||
+            requestedOptions.getVersion() > systemSupportedVersion
+        ) {
+            throw new IllegalArgumentException(
+                ("Unsupported proto version: %s. Was expecting at most version %s. " +
+                    "This library version supports up to version %s.").formatted(
+                        requestedOptions.getVersion(),
+                        supportedOptions.getVersion(),
+                        systemSupportedVersion
+                    )
+            );
+        }
+        if (requestedOptions.getGeneralizedStatements() && !supportedOptions.getGeneralizedStatements()) {
+            throw new IllegalArgumentException(
+                "The stream uses generalized statements, which are not supported. " +
+                "Either disable generalized statements or enable them in the supported options."
+            );
+        }
+        if (requestedOptions.getRdfStar() && !supportedOptions.getRdfStar()) {
+            throw new IllegalArgumentException(
+                "The stream uses RDF-star, which is not supported. Either disable" +
+                " RDF-star or enable it in the supported options."
+            );
+        }
+
+        checkTableSize("Name", requestedOptions.getMaxNameTableSize(), supportedOptions.getMaxNameTableSize(), 8);
+        checkTableSize("Prefix", requestedOptions.getMaxPrefixTableSize(), supportedOptions.getMaxPrefixTableSize());
+        checkTableSize(
+            "Datatype",
+            requestedOptions.getMaxDatatypeTableSize(),
+            supportedOptions.getMaxDatatypeTableSize()
+        );
+    }
+
+    private static void checkTableSize(String name, int size, int supportedSize, int minSize) {
+        if (size > supportedSize) {
+            throw new IllegalArgumentException(
+                "The stream uses a " +
+                name.toLowerCase() +
+                " table size of " +
+                size +
+                ", which is larger than the maximum supported size of " +
+                supportedSize +
+                "."
+            );
+        }
+        if (size < minSize) {
+            throw new IllegalArgumentException(
+                "The stream uses a " +
+                name.toLowerCase() +
+                " table size of " +
+                size +
+                ", which is smaller than the minimum supported size of " +
+                minSize +
+                "."
+            );
+        }
+    }
+
+    private static void checkTableSize(String name, int size, int supportedSize) {
+        checkTableSize(name, size, supportedSize, 0);
+    }
+
+    private static void checkLogicalStreamType(
+        Rdf.RdfStreamOptions options,
+        Rdf.LogicalStreamType expectedLogicalType
+    ) {
+        var logicalType = options.getLogicalType();
```

Comment:
Should we use `final var` in places like this? What is the convention in Java projects?

In C++ the `const` religion is a holy one, I know that. In Scala `val` is simply the default.

Unexpected mutations are horrible, but I'm not sure what is the community approach to this.

Reply:
doned